[RFC] ath79: dts LEDs default triggers

p-wassi opened the pull request:
On GL.iNet AR150 I came to the scenario, that the wifi LED was/is used as system status
indicator
https://github.com/openwrt/openwrt/blob/ab12913676b05c069f36eda827043c6b83e51dbe/target/linux/ath79/dts/ar9330_glinet_ar150.dts#L15-L17
and at the same time was assigned a default-trigger phy0:
https://github.com/openwrt/openwrt/blob/ab12913676b05c069f36eda827043c6b83e51dbe/target/linux/ath79/dts/ar9330_glinet_ar150.dts#L23-L26

After bootup, the LED is left with trigger = `none` in the sysfs, so although it is indicating bootup,
it does not show Wifi status later on.

First commit / question:
Do LEDs that are assigned as system status indicators additionally need a `default-state` or `default-trigger`? (From my observation on gl-ar150 the LED ends up with trigger `none` anyway)

Second commit:
To have this shared functionality (system status/bootup AND wifi), the trigger phy0tpt is added
in `/etc/config/system`for these two affected devices.

p-wassi commented:
> Using the default on trigger, the led is lit very early during the (kernel) boot process.

Ah, ok. I'll leave them.

> Same applies to the keep default-state.

Ok. So the 'keep' is actually just up to the moment the `diag.sh` takes over, right?
(I was only touching LEDs that were used/set by `diag.sh`)

> Don't add the phy trigger via 01_leds. If there is a collision, another led need to be hijacked
> for boot status indication. Another idea might be to store the initial set trigger and apply it
> after the boot is finished.

Hijacking another LED definitely is a way for new devices. Here, I'm more in favour of keeping the
behaviour we had in ar71xx.
What is your opinion on a change to /etc/diag.sh?
https://github.com/openwrt/openwrt/blob/48745221a8b4cdd18811a071387915e3ad1587ee/package/base-files/files/etc/diag.sh#L38-L43
Adding a check to `done)`, whether device-tree originally had another default-trigger set,
and if so, restore that value? (Just in case `$running` is not set)
So you could either have a 'running' indicator or a default-trigger after bootup.

mkresin commented:
> Ok. So the 'keep' is actually just up to the moment the diag.sh takes over, right?

Yes, keep what ever was set before. If this led used by diag.sh, diag.sh will change the status.

> Hijacking another LED definitely is a way for new devices. Here, I'm more in favour of keeping the
behaviour we had in ar71xx.

I'm more in favour of not doing the same mistakes done in ar71xx. But no strong opinion on this.

> Adding a check to done), whether device-tree originally had another default-trigger set,
and if so, restore that value? 

Yes, would work. But please send such a patch to the openwrt-devel mailing list, to get some more input.

p-wassi committed:
base-files: restore default-trigger for boot LEDs

For device-tree-devices only:
Restore the default-trigger for 'led-boot' LEDs if they were
previously set in device tree. If 'led-running' is set explicitly,
the 'led-boot' is turned off and 'led-running' on.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

mkresin commented:
Keep in mind, it will only work with really basic triggers. Something that requires an additional config, like the usbport trigger which requires to select the monitored ports, will not work.

p-wassi commented:
While currently debugging another issue on TP-Link WR740N-v4 (some LEDs stuck to 'on'):
For three devices, `linux,default-trigger = "heartbeat";` is set in the DTS.
However, the package `kmod-ledtrig-heartbeat` is not selected by default.
May I open another PR to remove those lines, or is there something I don't see right now?

ynezz commented:
I'm here with @mkresin just use free LED or just hijack the LED for failsafe/upgrade signalization.
